_targets.R

```diff
@@ -0,0 +1,295 @@
+)
+
+input_dir <- file.path(
+  Sys.getenv("GFH_DATA_DIR"),
+  "inputs"
+)
+
+nga_cod_gdb <- file.path(
+  Sys.getenv("AA_DATA_DIR"),
+  "public",
+  "raw",
+  "nga",
+  "cod_ab",
+  "nga_adm.shp.zip"
+)
+
+list(
+  # Track Files -------------------------------------------------------------
+  tar_target(
+    name = nga_aoi_kml_fp,
+    command = file.path(
+      input_dir,
+      "Nigeria AOIs.kml"
+    ),
+    format = "file"
+  ),
+  tar_target(
+    name = nga_pop_r_fp,
+    command = file.path(
+      input_dir,
+      "nga_pop",
+      "nga_general_2020.tif"
+    ),
+    format = "file"
+  ),
+  # Load Data ---------------------------------------------------------------
+  ## Base Data ####
+  # base data includes admin zones, rivers, basins,
+  ### Admin 0-2 ####
+  tar_target(
+    name = nga_adm,
+    command = read_shape_zip(path = nga_cod_gdb, layer = c(
+      "nga_admbnda_adm0_osgof_20190417",
+      "nga_admbnda_adm1_osgof_20190417",
+      "nga_admbnda_adm2_osgof_20190417"
+    )) %>%
+      set_names(c("adm0", "adm1", "adm2"))
+  ),
+  ### River MultiLines ####
+  tar_target(
+    name = nga_riv,
+    command = pull_dataset("741c6f20-6956-420d-aae4-37015cdd1ad4") %>%
+      get_resource(2) %>%
+      read_resource()
+  ),
+  # buffer rivers
+  tar_target(
+    name = nig_riv_b15k,
+    command = nga_riv %>%
+      filter(
+        NAME == "Niger"
+      ) %>%
+      st_transform("+proj=aeqd +lat_0=7.974 +lon_0=5.692") %>%
+      st_buffer(
+        dist = 15000
+      ) %>%
+      st_transform(
+        "EPSG:4326"
+      )
+  ),
+  ### Hydrobasins ####
+  # clipping basin levels 4 & 5 to NGA
+  tar_target(
+    name = basins_clipped,
+    command = load_clipped_hydrobasins(
+      gdb = file.path(input_dir, "hybas_af_lev01-12_v1c"),
+      lyr_names = c("hybas_af_lev04_v1c", "hybas_af_lev05_v1c"),
+      mask = nga_adm$adm0
+    )
+  ),
+  ## Google Inputs ####
+
+  ### Flood Extent Kmls ####
+  # note: not yet using this in the _targets pipeline - just adding as a simple target since it is
+  # in other wrangling/analysis scripts
+  tar_target(
+    name = gfh_coverage,
+    command = c("Nigeria_AOIs", "Nigeria_launched") %>%
+      map_dfr(\(lyr_name){
+        st_read(
+          dsn = nga_aoi_kml_fp,
+          layer = lyr_name
+        ) %>%
+          mutate(
+            coverage = lyr_name
+          )
+      })
+  ),
+  tar_target(
+    name = gfh_now,
+    command = gfh_coverage %>%
+      filter(coverage == "Nigeria_launched")
+  ),
+  ### Google real-time nowcast/forecast ####
+  
+  # read in full wb from google.
+  tar_target(
+    name = gauge_google_wb_full,
+    command = read_gauge_googlesheets(url = Sys.getenv("GFH_GAUGE_URL"))
+  ),
+
+  # remove any potential duplicate records by taking latest update_time_utc per date
+  tar_target(
+    name = gauge_google_wb,
+    command = gauge_google_wb_full %>%
+      imap(\(df, name_df){
+        gauge_realtime_names <- str_subset(name_df, "^hybas_.*$")
```

Comment:
yeah since this is an exploratory repo I think it might make sense to leave in the exploration process - as we can use all the data/information for vizuals etc and create new targets where the filtering is explained?